iOS/Collection View Management /CollectionViewDataSource.swift

```diff
@@ -319,16 +319,38 @@ extension CollectionViewDataSource {
 extension CollectionViewDataSource {
 
     /// Animates the first part of the animation cycle, applies the snapshot, then finishes the animation cycle.
+    @MainActor
     func apply(_ snapshot: SnapshotType,
                collectionView: UICollectionView,
-               animationCycle: AnimationCycle,
-               animatingDifferences: Bool = false) async {
+               animationCycle: AnimationCycle) async {
 
         await collectionView.animateOut(position: animationCycle.outToPosition,
                                         concatenate: animationCycle.shouldConcatenate)
 
-        await self.apply(snapshot, animatingDifferences: animatingDifferences)
+        // HACK: If we want to scroll to the end of the content, push the content way past the edge
+        // so we don't preload any cells undesired cells when the snapshop is applied.
+        if animationCycle.scrollToEnd {
+            collectionView.contentOffset = CGPoint(x: 999_999_999_999,
+                                                   y: 999_999_999_999)
+        }
```

Comment:
I would have used CGFloat.max, but that froze the app for some reason.

Reply:
lol.